// dao/dbConnect.js
/**
 * Created by apple on 2017/3/22.
 */
var mysql = require('mysql');
var connection = mysql.createConnection({
    host : '114.215.42.76' ,
    user : 'root' ,
    password : 'Liu123456' ,
    database : 'blog'
});

/**
 * 管理员类
 * @param admin
 * @constructor
 */
function Admin(admin) {
    this.adminanem = admin.adminname;
    this.password = admin.password;
}

Admin.login = function(adminname, callback){
    connection.query(
        "select * from admin where admin_name = '"+adminname+"'",
        function(err,res){
            console.log(res);
            callback(res,err);
        }
    )
}


/**
 * 发表文章接口
 * @param title 标题
 * @param text  文章内容
 * @param picture  标题图片
 * @param keyword  关键字
 * @param describe 简介
 */
Admin.sendArticle = function (title, text, picture, keyword, describe, callback){
    var sql = "INSERT INTO `article` (`article_title`, `article_text`, `article_picture`, `article_keyword`, `article_describe`) " +
        "VALUES ('"+title+"', '"+text+"', '"+picture+"', '"+keyword+"', '"+describe+"')";
    console.log(sql);
    connection.query(
        sql,
        function(err){
            // console.log(res);
            callback(err);
        }
    )
}


function Article(article){
    this.title = article.title;
    this.text = article.text;
    this.picture = article.picture;
    this.keyword = article.keyword;
    this.describe = article.describe
}

//分页获取文章列表
Article.getListTable = function(pageNumber, pageSize, callback){
    if(pageNumber == undefined){
        pageNumber = 1;
    }
    if(pageSize == undefined){
        pageSize = 5;
    }
    var pageEnd = pageNumber * pageSize;
    var pageBegin = pageNumber * pageSize - pageSize;
    var sql = 'SELECT * FROM article ORDER BY `id` desc LIMIT '+pageBegin+","+pageEnd;
    console.log(sql);
    connection.query(
        sql,
        function(err,res){
            callback(res,err);
        }
    )
}
Article.getIndexList = function (callback){
    var sql = "SELECT * FROM article where article_commend = '1'";
    console.log(sql);
    connection.query(
        sql,
        function(err,res){
            callback(res,err);
        }
    )
}
Article.getRightList = function (callback){
    var sql = "SELECT * FROM article where article_commend = '2'";
    console.log(sql);
    connection.query(
        sql,
        function(err,res){
            callback(res,err);
        }
    )
}
Article.getArticle = function (id, callback) {
    var sql = "SELECT * FROM article where id = '"+id+"'";
    console.log(sql);
    connection.query(
        sql,
        function(err,res){
            callback(res,err);
            var click = res[0].article_click;
            click = parseInt(click)+1;
            var sql = "UPDATE article SET article_click="+click+" where id = '"+id+"'";
            connection.query(sql);
        }
    )
}
Article.addClick = function (id, callback) {
    var sql = "SELECT * FROM article where id = '"+id+"'";
    connection.query(
        sql,
        function(err,res){

        }
    )

}
Article.sendComment = function (id,email,name,text,ip, ress) {
    var sql = "INSERT INTO article_comment (article_id, email, name, text, ip) VALUES ('"+id+"','"+email+"','"+name+"','"+text+"','"+ip+"')"
    console.log(sql);
    connection.query(
        sql,
        function(err,res){
            if(err == null){
                ress.json({msg:'恭喜你，成功发布了一条评论！'});
            }else{
                ress.json({msg:'糟糕，发现一个bug，如果在吃尝试后还是无法成功，请联系qq2388399752！'});
            }
        }
    )
}
Article.getComment = function (id,pageNumber,pageSize,callback) {
    var pageEnd = pageNumber * pageSize;
    var pageBegin = pageNumber * pageSize - pageSize;
    var sql = "SELECT * FROM article_comment WHERE article_id = '"+id+"' ORDER BY `id` desc LIMIT "+pageBegin+","+pageEnd;
    console.log(sql);
    connection.query(
        sql,
        function(err,res){
            callback(res, err);
        }
    )
}



function Website(website) {
    this.id = id;
    this.website_title = website_title;
    this.website_logo = website_logo;
    this.website_realmname = website_realmname;
    this.website_keyword = website_keyword;
    this.website_describe = website_describe;
    this.website_tel = website_tel;
    this.website_qq = website_qq;
    this.website_email = website_email;
    this.website_address = website_address;
    this.website_bottom = website_bottom;
    this.website_status = website_status;
    this.website_aboutme = website_aboutme;
}
Website.getWebsite = function (id, callback) {
    var sql = "select * from website where id = '"+id+"'";
    console.log(sql);
    connection.query(
        sql,
        function(err,res){
            callback(res,err);
        }
    )
}







function Message(message) {
    this.id = message.id;
    this.message_name = message_name;
    this.message_time = message_time;
    this.message_value = message_value;
    this.message_fid = message_fid;
    this.message_replaynum = message_replaynum;
}
Message.getMessage = function (pageNumber, pageSize, callback) {
    if(pageNumber == undefined){
        pageNumber = 1;
    }
    if(pageSize == undefined){
        pageSize = 5;
    }
    var pageEnd = pageNumber * pageSize;
    var pageBegin = pageNumber * pageSize - pageSize;
    var sql = 'SELECT * FROM message ORDER BY `id` desc LIMIT '+pageBegin+","+pageEnd;
    connection.query(
        sql,
        function(err,res){
            console.log(res);
            callback(res,err);
        }
    )
}
Message.addMessage = function (name, msg, fid, ip, ress) {
    var sql = "INSERT INTO message (message_name, message_value, message_fid, message_ip)VALUES ('"+name+"', '"+msg+"', '"+fid+"', '"+ip+"')";
    connection.query(
        sql,
        function(err,res){
            if(fid+"" != "0"){
                var sql = "SELECT message_replaynum FROM message WHERE id = '"+fid+"'";
                connection.query(
                    sql,
                    function(err,res){
                        var message_replaynum = res[0].message_replaynum + 1;
                        var sql = "UPDATE message SET message_replaynum = '"+message_replaynum+"' WHERE id='"+fid+"'";
                        connection.query(
                            sql,
                            function(err,res){
                                console.log(err);
                                if(err != null){
                                    ress.json({msg:"发布成功!"});
                                }else{
                                    ress.json({msg:"发布失败!"});
                                }
                            })
                    })

            }else{
                if(err+"" === "null"){
                    ress.json({msg:"发布成功!"});
                }else{
                    ress.json({msg:"发布失败!"});
                }
            }
        }
    )
}




exports.admin = Admin;
exports.article = Article;
exports.website = Website;
exports.message = Message;
